test/runtests.jl

```diff
@@ -43,6 +53,43 @@ end
 
     mar5 = ArrayMargins(X, DimIndices([[2, 1]]))
     @test mar5.am[1] == X'
+
+    # Test for overlapping dimension case, as well as Float64 type
+    di = DimIndices([[5,2,4], [1,4,3], [2,5,1], [4,2]])
+    X = convert(Array{Float64}, reshape(repeat(1:20, 15), 3, 2, 5, 2, 5))
+    mar6 = ArrayMargins(X, di)
+    @test typeof(mar6) == ArrayMargins{Float64}
+    @test length(mar6) == 4
+    @test ndims(mar6) == 5
+    @test size(mar6) == (3, 2, 5, 2, 5)
+
+    # Consistency check
+    @test isconsistent(mar6)
+    mar6_p = proportion_transform(mar6)
+    @test sum.(mar6_p.am) ≈ fill(1., 4)
+    @test check_margin_totals(mar6)
+
+    # Check we can catch margin inconsistency
+    mar7 = deepcopy(mar6)
+    mar7.am[2][1,2,4] += 1 #augment one value by one
+    @test !isconsistent(mar7)
+    @test_warn r"Margin totals do not match" check_margin_totals(mar7)
+
+    # Check we can achieve proportion consistency even if margin totals are not consistent
+    mar8 = deepcopy(mar6)
+    mar8.am[1] .*= 2.5 #scale
+    @test !isconsistent(mar8; tol = sqrt(eps(Float64)))
+    @test_warn r"Margin totals do not match" check_margin_totals(mar8)
+    mar8_p = proportion_transform(mar8)
+    @test isconsistent(mar8_p; tol = sqrt(eps(Float64)))
+    @test_nowarn check_margin_totals(mar8_p; tol = sqrt(eps(Float64)))
+
+    # Test we catch mismatched lengths of repeated dimensions
+    target_13 = fill(3, (2,4))
+    target_23 = fill(2, (3,4))
+    di = DimIndices([[1,3], [2,1]]) #incorrect should be [[1,3], [2,3]]
+    m = ArrayMargins([target_13, target_23], di)
+    @test_throws DimensionMismatch size(m)
```

Comment:
This will change if we put the check in the constructor; then the @test_throws DimensionMismatch should be one line higher 👍